enhancements/network/egress-qos.md

```diff
@@ -0,0 +1,187 @@
+---
+title: OVN Pods Egress DSCP QoS
+authors:
+  - "@oribon"
+reviewers:
+  - "@trozet"
+  - "@danwinship"
+  - "@tssurya"
+approvers:
+  - TBD
+creation-date: 2022-02-16
+last-updated: 2022-02-27
+status: implementable
+---
+
+# OVN Pods Egress DSCP QoS
+
+## Summary
+
+Not all traffic has the same priority, and when there is contention for bandwidth, there should be a mechanism for objects outside the cluster to prioritize the traffic.
+To enable this, we will use Differentiated Services Code Point (DSCP) which allows us to classify packets by setting a 6-bit field in the IP header, effectively marking the priority of a given packet relative to other packets as "Critical", "High Priority", "Best Effort" and so on.
+
+By introducing a new CRD `EgressQoS`, users could specify a DSCP value for packets originating from pods on a given namespace heading to a specified CIDR.
+The CRD will be Namespaced, with one resource allowed per namespace.
```

Comment:
The API should probably allow for namespace selector or pod selectors to allow for a more granular control or do we want this to be purely a per-namespace thing?

Reply:
I think it would be clearer having the CRD namespace-scoped and maybe a bit easier to implement that way. We'll achieve per-namespace egress dscp both ways, but won't have to manage/watch namespaces.
I believe pod selectors is a field we could add later as part of the egress rule if we'd want.